### PR TITLE
Create util to verify KZG commitment by generation and comparison

### PR DESCRIPTION
**Pull request by litt3**
## Why are these changes needed?

- This PR is part of https://github.com/Layr-Labs/eigenda/issues/1000
- It includes a utility to verify kzg commitments by regenerating them, and comparing vs the claimed commitment
- The optimized version of verification using fiat shamir transformation will be included in a future PR

## Checks

- [X] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- [ ] I've checked the new test coverage and the coverage percentage didn't drop.
- Testing Strategy
   - [X] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: api/clients/verification/verification_utils_test.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+	kzgConfig       *kzg.KzgConfig
+	kzgVerifier     *verifier.Verifier
+	testRandom      *random.TestRandom
+	srsNumberToLoad uint64
+)
+
+func setup() {
+	log.Println("Setting up suite")
```
Comment (cody-littley) [inline api/clients/verification/verification_utils_test.go:26]: nit, can this be removed?

File: api/clients/verification/verification_utils_test.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+func teardown() {
+	log.Println("Tearing down")
+}
+
+func TestComputeAndCompareKzgCommitmentSuccess(t *testing.T) {
+	kzgVerifier, err := verifier.NewVerifier(kzgConfig, nil)
+	assert.NotNil(t, kzgVerifier)
+	assert.Nil(t, err)
```
Comment (cody-littley) [inline api/clients/verification/verification_utils_test.go:64]: Personally, I prefer using `require`. Assert will allow the remainder of the test to run and can sometimes permit odd errors. `require` causes the test to abort immediately when the first failure is detected.
```suggestion
	require.NotNil(t, kzgVerifier)
	require.Nil(t, err)
```

File: api/clients/verification/verification_utils_test.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+	commitment, err := GenerateBlobCommitment(kzgVerifier, gettysburgAddressBytes)
+	assert.NotNil(t, commitment)
+	assert.Nil(t, err)
+
+	// make sure the commitment verifies correctly
+	err = GenerateAndCompareBlobCommitment(
+		kzgVerifier,
+		commitment,
+		gettysburgAddressBytes)
+	assert.Nil(t, err)
```
Comment (cody-littley) [inline api/clients/verification/verification_utils_test.go:75]: This will yield a better error message when checking for errors.
```suggestion
	require.NoError(t, err)
```

File: api/clients/verification/verification_utils_test.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+func TestMain(m *testing.M) {
+	setup()
+	result := m.Run()
+	teardown()
+	os.Exit(result)
+}
+
+func teardown() {
+	log.Println("Tearing down")
+}
```
Comment (cody-littley) [inline api/clients/verification/verification_utils_test.go:59]: Now that the test is functional, is this needed?

File: api/clients/verification/verification_utils_test.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
... so conceived, and so dedicated, can long endure. We are met on a great battle-field of that war. We have come to dedicate a portion of that field, as a final resting-place for those who here gave their lives, that that nation might live. It is altogether fitting and proper that we should do this. But, in a larger sense, we cannot dedicate, we cannot consecrate—we cannot hallow—this ground. The brave men, living and dead, who struggled here, have consecrated it far above our poor power to add or detract. The world will little note, nor long remember what we say here, but it can never forget what they did here. It is for us the living, rather, to be dedicated here to the unfinished work which they who fought here have thus far so nobly advanced. It is rather for us to be here dedicated to the great task remaining before us—that from these honored dead we take increased devotion to that cause for which they here gave the last full measure of devotion—that we here highly resolve that these dead shall not have died in vain—that this nation, under God, shall have a new birth of freedom, and that government of the people, by the people, for the people, shall not perish from the earth."))
```
Comment (cody-littley) [inline api/clients/verification/verification_utils_test.go:17]: Would it be better to generate random bytes during each test (in case there is an edge case that these bytes don't trigger)?

File: api/clients/verification/verification_utils_test.go [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+	indexToModify := testRandom.Intn(len(inputBytes))
+	inputBytes[indexToModify] = inputBytes[indexToModify] + 1
+}
+
+func TestMain(m *testing.M) {
+	setup()
+	result := m.Run()
+	teardown()
+	os.Exit(result)
+}
```
Comment (cody-littley) [inline api/clients/verification/verification_utils_test.go:55]: Why is this necessary?

**Review by cody-littley — APPROVED** [review]

File: api/clients/verification/verification_utils_test.go (hunk 4 above)
Comment (litt3) [reply api/clients/verification/verification_utils_test.go]: removed

File: api/clients/verification/verification_utils_test.go (hunk 5 above)
Comment (litt3) [reply api/clients/verification/verification_utils_test.go]: Good point, done

File: api/clients/verification/verification_utils_test.go (hunk 6 above)
Comment (litt3) [reply api/clients/verification/verification_utils_test.go]: It's not, removed

File: api/clients/verification/verification_utils_test.go (hunk 1 above)
Comment (litt3) [reply api/clients/verification/verification_utils_test.go]: Done

File: api/clients/verification/verification_utils_test.go (hunk 2 above)
Comment (litt3) [reply api/clients/verification/verification_utils_test.go]: Done

File: api/clients/verification/verification_utils_test.go (hunk 3 above)
Comment (litt3) [reply api/clients/verification/verification_utils_test.go]: Done

**Review by anupsv — APPROVED** [review]

File: api/clients/v2/verification/verification_utils.go [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+	kzgVerifier *verifier.Verifier,
+	claimedCommitment *encoding.G1Commitment,
+	blobBytes []byte) error {
+
+	computedCommitment, err := GenerateBlobCommitment(kzgVerifier, blobBytes)
+	if err != nil {
+		return fmt.Errorf("compute commitment: %w", err)
+	}
+
+	if claimedCommitment.X.Equal(&computedCommitment.X) &&
```
Comment (anupsv) [inline api/clients/v2/verification/verification_utils.go:51]: i was going to say we'd need to check if claimedCommitment is on the curve but since its checked against what's computed locally, its fine.

File: api/clients/v2/verification/verification_utils.go [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+	"github.com/consensys/gnark-crypto/ecc"
+	"github.com/consensys/gnark-crypto/ecc/bn254"
+
+	"github.com/Layr-Labs/eigenda/encoding/kzg/verifier"
+	"github.com/Layr-Labs/eigenda/encoding/rs"
+)
+
+// GenerateBlobCommitment computes a kzg-bn254 commitment of blob data using SRS
+func GenerateBlobCommitment(
+	kzgVerifier *verifier.Verifier,
```
[the remark this answers is not in the document]
Comment (litt3) [reply api/clients/v2/verification/verification_utils.go:15]: Unconscious hesitancy to modify an existing utility, I suppose.

You're right, it's where the methods belong. I will fix this in an upcoming PR
